### ISSUE TITLE
[BUG] The builder can get stuck when digging blocks if they're in a tricky position.

### ISSUE THREAD

**Issue by fulmoonless** [ldtteam/minecolonies#10124]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **beta** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [X] Single Player
- [ ] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

1.20.1-1.1.629

### Structurize Version

1.20.1-1.0.742

### Related Mods and their Versions

Minecolonies Offical WinterColonies-1.11

### Current Behavior

I'm not sure how to describe this, so I'll just upload a video.
https://youtu.be/1_aagAFlOmk

### Expected Behavior

Don’t get stuck."

### Reproduction Steps

1. Build a Functional Fantasy university.
2. Request a building in a similar location?

### Logs

No crash

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by MotionlessTrain** [comment]
From what I noticed myself, and from what I gathered whilst helping someone on discord, it looks like the builder is not allowed to emergency teleport themselves away, when they build/dig themselves in a corner.
So at night, they are at risk for mobs when they dig out a basement, if you don't recall them to their builder's hut, the town hall or their residence manually during sunset, or they don't go back to the builder's hut when they need an item (which can cause them to never request that item)
The latter of those has been happening in 1.20.1-1.1.683

**Comment by Talyda** [comment]
Gonna add on, I noticed this yesterday myself. Builder was building the fortress mine, the one with the built in guard towers, and would not teleport out at all.
Glad you mentioned it Train, cos I forgot between then and now =P

**Comment by Estyria** [comment]
I have also noticed they will not teleport.

**Comment by OracleMask** [comment]
I have had many many instances of builders getting stuck in a hole of a build they are working on, ranging from mines to warehouses to mystic sites.

**Comment by Raycoms** [comment]
> Gonna add on, I noticed this yesterday myself. Builder was building the fortress mine, the one with the built in guard towers, and would not teleport out at all. Glad you mentioned it Train, cos I forgot between then and now =P

How long would they stay stuck there? I tried a couple of things and it would take 5-10 minutes and the builder would tp just fine

**Comment by Estyria** [comment]
Ray I've seen them stay indefinitely.

**Comment by Talyda** [comment]
> How long would they stay stuck there? I tried a couple of things and it would take 5-10 minutes and the builder would tp just fine

Yeah, as Esty said, they stayed there 'til I recalled them (multiple in game days)
It was replicable with at least the fortress mine (I had it happen more than once in that test world)

**Comment by Talyda** [comment]
I actually have a whole heap of appointments this morning, so can't really hop on the server, so I will start a test game while I have breakfast =) Just double check!

**Comment by Talyda** [comment]
NATURALLY they both worked today.
I tested multiple times the other day - failed each time. But a couple people also reported it with Caledonia guard towers the other weeks, and I did 5 towers in a test world and they were all fine

**Comment by Vwhbj2003** [comment]
I'm currently having this problem. I've tried down grading to 1.1.515 but the game won't load. It gives me an invalid player data pop up and kicks me over to a server list. I play single player. I've also tried updating to the newest version but nothing changed. Is there a more recent version without this bug?
